imports/util/recordParser.js

```diff
@@ -44,11 +51,129 @@ export function parseLengthOfTreatment(stringValue) {
     stringValue = stringValue.toLowerCase();
     if (stringValue.includes("years")) {
       return number * 365;
-    } else if(stringValue.includes("months")) {
+    } else if (stringValue.includes("months")) {
       return number * 30;
     } else {
       return 0;
     }
   }
   return 0;
-}
\ No newline at end of file
+}
+
+function getRecommendations(header) {
+  const relatedWords = header.split(" ");
+  const headerValues = settings.public.RECORD_TEMPLATE;
+  let recommendations = [];
+  relatedWords.forEach(word => {
+    const re = new RegExp(word, "i");
+    const result = headerValues.filter(value => {
+      return value.display_name.match(re);
+    });
+    recommendations = recommendations.concat(result);
+  });
+
+  return recommendations;
+}
+
+/**
+ *
+ * @param {string} header
+ * @returns {id: string, value(parsed value): string, valid: boolean, recommended: string[]} ParsedHeader
+ */
+function parseHeader(header) {
+  //set an uniqid so we can find the cells using the old header in case of updating header.
+  //read into the settings column.
+  //if no match, try to break header into
+  //array of words and find the column that
+  //include all the words.toLower().
+  //if still no match,
+  //return the header converted to snake case, then set valid to false
+  //then find the column that include the most of the words,
+  //ranked from most likely to least.
+  const id = UniqueId();
+  const headerValues = settings.public.RECORD_TEMPLATE;
+  const re = new RegExp(header, "i");
+  const filteredHeaders = headerValues.filter(headerValue => {
+    if (headerValue.display_name === "Birthday" && header === "Birthday") {
+      console.log(headerValue, header);
+    }
+
+    return headerValue.display_name.match(re);
+  });
+  let valid = null;
+  let parsedHeader = null;
+  let recommended = [];
+  // console.log(filteredHeaders);
+  const snakedCaseHeader = changeCase.snakeCase(header);
+  if (filteredHeaders.length === 0) {
+    valid = false;
+    parsedHeader = snakedCaseHeader;
+    recommended = getRecommendations(header);
+  } else {
+    valid = true;
+    parsedHeader = filteredHeaders[0].display_name;
+  }
+
+  // console.log({ id, value: parsedHeader, valid, recommended });
+  return {
+    id,
+    value: parsedHeader,
+    valid,
+    recommended,
+    updatedValue: snakedCaseHeader
+  };
+}
+
+/**
+ *
+ * @param {string} value
+ * @param {ParsedHeader} parsedHeader
+ * @returns {value: string, valid: boolean, parser: function}
+ */
+function parseCell(value, parsedHeader) {
+  const valid = value ? true : false;
+
+  return {
+    id: parsedHeader.id,
+    value,
+    valid,
+    parsedHeader,
+    parser: null
```

Comment:
@nerocui - for the parser, you wanted to return a function. What is that function that you need?